From bd63b4eb5a7d9d1d34849ea14d23f4a4d2d4f93c Mon Sep 17 00:00:00 2001
From: zachary <zhangzg@marvell.com>
Date: Tue, 19 Sep 2017 18:10:10 +0800
Subject: [PATCH 2010/2241] fix: clock: mvebu: armada-37xx-pm-clk: correct CPU
 TBG clock source in PMU

When CPU is 1200MHZ, TBG clocks' source used to be
TBG-A_P, which is not correct.
The patch is to set their socure clock to TBG-B_S.

Change-Id: I1be72474549cddfbc0698e778d6ca3c3fea3ec04
Signed-off-by: zachary <zhangzg@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/44503
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Victor Gu <xigu@marvell.com>
---
 drivers/clk/mvebu/armada-37xx-pm-clk.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/clk/mvebu/armada-37xx-pm-clk.c b/drivers/clk/mvebu/armada-37xx-pm-clk.c
index 4da10a7..50a94b2 100644
--- a/drivers/clk/mvebu/armada-37xx-pm-clk.c
+++ b/drivers/clk/mvebu/armada-37xx-pm-clk.c
@@ -139,7 +139,7 @@ struct armada_a3700_clk_pm_info {
  *   600M                 600M              300M             240M            200M
  */
 static struct armada_a3700_clk_pm_info a3700_pm_info[] = {
-	{.cpu_freq_max = 1200, .clk_sel = CLK_SEL_TBG, .tbg_sel = TBG_A_P, .divider = {1, 2, 4, 6} },
+	{.cpu_freq_max = 1200, .clk_sel = CLK_SEL_TBG, .tbg_sel = TBG_B_S, .divider = {1, 2, 4, 6} },
 	{.cpu_freq_max = 1000, .clk_sel = CLK_SEL_TBG, .tbg_sel = TBG_B_S, .divider = {1, 2, 4, 5} },
 	{.cpu_freq_max = 800,  .clk_sel = CLK_SEL_TBG, .tbg_sel = TBG_A_P, .divider = {1, 2, 3, 4} },
 	{.cpu_freq_max = 600,  .clk_sel = CLK_SEL_TBG, .tbg_sel = TBG_A_P, .divider = {2, 4, 5, 6} },
-- 
2.7.4

